include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h

```diff
@@ -106,20 +109,25 @@ struct __subgroup_radix_sort
     static constexpr uint16_t __counter_buf_sz = __wg_size * __bin_count + 1; //+1(init value) for exclusive scan result
 
     template <typename _T, typename _Size>
-    bool
+    auto
     __check_slm_size(sycl::queue __q, _Size __n)
     {
         assert(__n <= 1 << 16); //the kernel is designed for data size <= 64K
 
+        const auto __req_slm_size_counters = __counter_buf_sz * sizeof(uint32_t);
+
         // Pessimistically only use half of the memory to take into account memory used by compiled kernel
         const ::std::size_t __max_slm_size =
-            __q.get_device().template get_info<sycl::info::device::local_mem_size>() / 2;
+            (__q.get_device().template get_info<sycl::info::device::local_mem_size>() - __req_slm_size_counters) / 2;
```

Comment:
I don't think we want to remove `__req_slm_size_counters` here.  We are incorporating that size below in the conditionals instead (lines 126, 128).

Reply:
dont catch....

We consider the total SLM size as `device::local_mem_size/2`.. right?
After that we checks is it enough for values and counters or just for counters... or  - not for both

Reply:
```suggestion
            __q.get_device().template get_info<sycl::info::device::local_mem_size>() / 2;
```


Further explanation:

` const ::std::size_t __max_slm_size = (__q.get_device().template get_info<sycl::info::device::local_mem_size>() - __req_slm_size_counters) / 2;`
gets the max SLM size, subtracts the calculated required size the counters would require, and then divides by 2.  I think instead it should just get the max SLM size, and divide by 2.

`if(__req_slm_size_val + __req_slm_size_counters <= __max_slm_size)` 
Checks if both the values and the counters fit into max memory (but as written in the PR, the counter size has already been removed).

`if(__req_slm_size_counters <= __max_slm_size)`
Checks if only the counters fit into max memory (but as written in the PR, the counter size has already been removed).

Reply:
fixed